neo4j/src/main/java/org/hibernate/ogm/datastore/neo4j/embedded/impl/EmbeddedNeo4jSchemaDefiner.java

```diff
@@ -59,6 +64,42 @@ protected void createUniqueConstraintsIfMissing(DatastoreProvider provider, List
 		EmbeddedNeo4jDatastoreProvider neo4jProvider = (EmbeddedNeo4jDatastoreProvider) provider;
 		GraphDatabaseService neo4jDb = neo4jProvider.getDatabase();
 		Transaction tx = neo4jDb.beginTx();
+		if ( tx instanceof TopLevelTransaction ) {
```

Comment:
The code can be removed. It is for debug and code investigations only.

Reply:
I don't think we should care about the Transaction specific implementation.

Reply:
Yes, @DavideD . I will remove it. The code is not needs for 'ready for merge' PR. The code is for debug and investigation only.